library/core/src/slice/iter.rs

```diff
@@ -2222,82 +2219,70 @@ impl<'a, T, const N: usize> Iterator for ArrayWindows<'a, T, N> {
 
     #[inline]
     fn next(&mut self) -> Option<Self::Item> {
-        if self.num == 0 {
-            return None;
+        if let Some(ret) = self.v.first_chunk() {
+            self.v = &self.v[1..];
+            Some(ret)
+        } else {
+            None
```

Comment:
```suggestion
        self.v.first_chunk().inspect(|_| self.v = &self.v[1..])
```

Reply:
No, please don't. That's clever, but way too obfuscated for my taste.